### ISSUE TITLE
Incorrect disco#info item-not-found response on private room

### ISSUE THREAD

**Issue by ge0rg** [matrix-org/matrix-bifrost#116]
Using my xmpp identity on the official bridge (`@_xmpp_georg=40yax.im:matrix.org`), I'm a member in a private room, let's call it `#mysecretchannel:matrix.org`.

When my XMPP client requests a disco#info from this room, I expect a proper MUC info and description. Instead, I get an `item-not-found` error:

```xml
SEND: 
<iq id="34ce88cd0cd144efab2b73de544b5d46" to="#mysecretchannel#matrix.org@matrix.org" type="get">
  <query xmlns="http://jabber.org/protocol/disco#info"/>
</iq>

RECV: 
<iq id="34ce88cd0cd144efab2b73de544b5d46" to="georg@yax.im/poezio-IS8H" xml:lang="en" from="#mysecretchannel#matrix.org@matrix.org" type="error">
  <error type="cancel" code="404">
    <item-not-found xmlns="urn:ietf:params:xml:ns:xmpp-stanzas"/>
  </error>
</iq>
```

**Comment by Half-Shot** [comment]
@ge0rg this might be fixed now that we use a different method of getting the information.

**Comment by ge0rg** [comment]
I wasn't able to join or query that room until today. Today I realized that the room forgot that my xmpp entity is a room member. After inviting myself I successfully joined it today around 11:55 CEST and it used to work for a while until the afternoon.

Right now, I can't join the room and I see the same IQ response. Also it looks like the room once again forgot that I'm a member?! Is it treating an xmpp unavailable presence as an attempt to stop being a member?
